Clang warning on Array resize call

Here's the code:
```
function main() {
    let mutable x = [0]
    let y = x.pop()!
}
```

Here's the clang warning:
```
runtime/Builtins/Array.h:231:9: warning: ignoring return value of function declared with 'nodiscard' attribute [-Wunused-result]
        resize(size() - 1);
        ^~~~~~ ~~~~~~~~~~
build/test.cpp:20:29: note: in instantiation of member function 'JaktInternal::Array<long long>::pop' requested here
        const i64 y = (((x).pop()).value());
```

--- a/runtime/AK/StringBuilder.cpp
+++ b/runtime/AK/StringBuilder.cpp
@@ -77,7 +77,7 @@ StringView StringBuilder::string_view() const
 
 void StringBuilder::clear()
 {
-    m_buffer.resize(0);
+    static_cast<void>(m_buffer.resize(0));
 }
 
 ErrorOr<void> StringBuilder::try_append_code_point(u32 code_point)

--- a/runtime/Builtins/Array.h
+++ b/runtime/Builtins/Array.h
@@ -245,7 +245,7 @@ public:
         if (is_empty())
             return {};
         auto value = move(at(size() - 1));
-        resize(size() - 1);
+        static_cast<void>(resize(size() - 1));
         return value;
     }
 

